cpp/tensorrt_llm/batch_manager/kvCacheManager.cpp

```diff
@@ -778,6 +777,10 @@ BlockPtr WindowBlockManager::getFreeBlock(
     ++mAllocTotalBlocks;
     if (!block->getUniqueTokens().empty() && canOffload && mEvictionPolicy->getNumFreeBlocks(kSecondaryLevel) > 0)
     {
+        // Since `claimBlock` resets link between parentBlock and block
+        // we need to save origin parent block.
+        const auto parentBlock = block.getPrevBlock(); 
```

Comment:
I have not tested offloading to a secondary pool as I don't use such a feature in prod.